### ISSUE TITLE
Why those permissions?

### ISSUE THREAD

**Issue by HenkiedeHenkie** [TwidereProject/Twidere-Android#889]
Started to use Twidere on my Android device. However, I'm wondering why the app needs so much permissions when signing in with a Twitter account. Why needs Twidere to be able to see my Tweets, send Tweets and read my private messages? For an app that promises to be without ads and respect user's privacy, this is rather strange...

**Comment by orithena** [comment]
Ummm... are you serious?

The Twidere _app_ on your _phone_ needs these permissions to be of any use to you, i.e. to display your timeline to _you_, to allow _you_ to send tweets and to display and send _your_ direct messages. It needs these permissions to fulfill the function it is designed for, i.e. _being a Twitter client_.

I'm at a loss what you thought Twidere was _for_?

It's not the case that the people who wrote the Twidere app would be able to see "your Twitter". When you login using your Twitter credentials, a secret key is negotiated that allows the Twidere app to login to Twitter in the future without asking you again for your credentials. This secret key stays on your phone and your phone alone (and the backup copy in your Google account, if you didn't disable it... and probably also on the NSA servers, but if you're concerned about that, you wouldn't use Twitter). 

Unless, of course, the people behind Twidere are sneaky enough to insert some code into the Twidere app that sends them that secret key. But that's what Open Source is for. You're welcome to check Twideres source code for any traces of such activity. (Hint: There ain't no such malicious activity. Other people would have seen them already and Twidere would not be in the Play Store any more as a consequence.)

You are very right in asking why any funny web app should be able to read your direct messages or be able to spam your followers with any tweets... but a fully-fledged Twitter client _does_ need all of these permissions to be of any use to you.

**Comment by mariotaku** [comment]
TL;DR: Those permissions are used by Twidere for accessing those data from Twitter, and present them for you. Twidere itself won't use these data for any other methods.

Full version: See @orithena 's comment.
